### PR TITLE
[pack] Added build yml and introduced versioning

### PR DESCRIPTION
**Pull request by yojagad**
- Added yml that builds and runs tests. Artifacts are generated only if a PR has [pack] keyword or during CI.
- Artifacts are signed only if it's a release branch.
- Versioning assumes major.minor.patch.build for dev branch and major.minor.patch for release branch(es). Dev branch's version.txt always points to the future version to be merged into release branch and released to myget and/or nuget.

### CODE REVIEW

File: build/version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+1.0.0
```
Comment (yojagad) [inline build/version.txt]: This always points to the next version to be released. After every release, minor version is updated here and merged into dev branch (there will be automation for this).

Comment (pragnagopa) [reply build/version.txt]: Version should be coming from `.csproj` similar to other extensions and webjobs sdk: https://github.com/Azure/azure-webjobs-sdk/pull/2610/files

**Review by pragnagopa — COMMENTED** [review]
Added a comment

**Comment by yojagad** [comment]
Update where version is got from.

File: src/WebJobs.Extensions.RabbitMQ.csproj [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <SignAssembly>false</SignAssembly>
     <AssemblyOriginatorKeyFile>PublicKey.snk</AssemblyOriginatorKeyFile>
     <DelaySign>true</DelaySign>
-    <Version>0.2.0$(VersionSuffix)-beta</Version>
+    <Version>1.0.0</Version>
```
Comment (pragnagopa) [inline src/WebJobs.Extensions.RabbitMQ.csproj]: This should be `0.2.0$(VersionSuffix)-beta` . `beta` is needed until we GA. `VersionSuffix` is needed to generate packages with build number right?

Comment (yojagad) [reply src/WebJobs.Extensions.RabbitMQ.csproj]: We can probably discuss this more if needed but what I'm thinking is, Dev branch is pointing to a version we are working towards. In this case, when a release is cut off by merging to release branch, 1.0.0 is what goes to nuget. After cutting it off, minor version in Dev branch is updated (1.1.0, 1.2.0, etc) indicating that's the version we are working towards.

All of what I described above would be automated of course.

Comment (yojagad) [reply src/WebJobs.Extensions.RabbitMQ.csproj]: Also, if dev branch needs to generate package, we do it with [pack] and that would generate with build number. Format being major.minor.patch.build (1.0.0.1, 1.0.0.2, etc). Dev packages won't be signed though.

Upgrading any of major, minor or patch version would reset this build number back to 1

Comment (pragnagopa) [reply src/WebJobs.Extensions.RabbitMQ.csproj]: I think we need to consistent across repos with how we version packages. @brettsam might be able to provide better feedback here.

Ideally, using build numbers in the packages directly without us maintaining a counter will help co-relate a package to build number - easy to track which build generated the package.

Regardless - do bring `-beta`  back until we are ready to GA.

Comment (fabiocav) [reply src/WebJobs.Extensions.RabbitMQ.csproj]: We do want to keep the version suffix. Moving to 1.0.0 makes sense, but let's indeed keep the `beta` flag in the meantime.

**Review by fabiocav — APPROVED** [review]

**Comment by yojagad** [comment]
I added the -beta prefix. But I'm don't think that adding $(VersionSuffix) is necessary since I already have logic in place for non-release builds to have incremental build numbers here https://github.com/Azure/azure-functions-rabbitmq-extension/blob/3d9ac9ba7adaad4d3c958677b77add4115354db6/build/DetermineVersionAndSigning.ps1 

azure-pipelines.yml in turn uses the versionNumber listed as highlighted here
image

**Review by pragnagopa — APPROVED** [review]
Thanks for adding more details. Makes sense. Please make sure artifacts generated are as expected before merging this PR.

**Pull request merged**